w06-ruby-on-rails/d4-dawn-intro-rails/README.md

```diff
@@ -0,0 +1,195 @@
+
+**M**odel - The model refers to the data objects that we use. It's the object oriented approach to design. The data in our database will be the most common type of object that we'll put there.
+
+**V**iew - The view is the Presentation layer. It's what the user sees and interacts with, essentially the web pages. The HTML, the CSS and the JavaScript. The controller processes and responds to user events, such as clicking on links and submitting forms.
+
+**C**ontroller - The controller will make decisions based on the request and then control what happens in response. It controls the interaction with our models and with our views.
+
+![MVC Diagram](http://elibildner.files.wordpress.com/2012/06/screen-shot-2012-06-05-at-2-12-18-am.png)
+
+## Railstaurant Metaphor (10m)
+The **client** is a customer eating in the restaurant, **Rails** is the kitchen, the **request** is the order made, the **router** is the waiter, the **controller** is a chef, the **model** is a recipe, the **database** is the giant walk-in refrigerator with ingredients, the **view** is plating the dish to look pretty, the **response** with a file is a waiter finally serving the dish to the customer.
+
+## Setup (10m)
+
+### How to create a Rails project
+
+`rails new NAME_OF_APP`
+
+At this point Rails will generate a bunch of directories and files. Once it is done, it runs
+`bundle install`, by which it's telling Bundler (a gem) to install all of the gems that might be missing.
+
+## Gems
+Gems are like the modules you installed in Express using `npm install <module name>`. You have to put any gem you want to use in your Gemfile. You have to run `bundle` from the Terminal anytime you change your Gemfile. Your `rails s(erver)` needs to be restarted after any changes to your Gemfile.
+
+### Bundler
+
+Bundler is a separate gem from Rails, and can be used outside of
+Rails, but Rails is going to depend on it to manage the RubyGems that the application needs. The first thing that you need to know is that there are two files that matter to bundler: `Gemfile` and `Gemfile.lock`. Gemfile contains configuration information about what
+gems are part of our project, it is similar to the `package.json` file in Node.
+
+Bundler looks at the `Gemfile` loading all the included gems in addition to each's dependencies. It then generates a manifest file that is stored in Gemfile.lock. **Never** edit Gemfile.lock!
+
+So how do you tell Bundler to take your `Gemfile` and turn it into `Gemfile.lock`? Run: `bundle install`. When we create a Rails application it will run this command for us unless we specify otherwise.
+
+## Running Rails (5m)
+
+- Create a new Rails app with `rails new railsFun --skip-activerecord`
+    - the last flag tells the application to not use `activerecord`
+    - `activerecord` is our ORM that manages our Models and our database, we'll learn more about it this afternoon
+- `cd` into your `railsFun` folder and run
+- run `rails server` or just `rails s` and see what happens
```

Comment:
Should we mention that they should keep rails server running in a separate tab or do you think their experience with nodemon carries over sufficiently?